detect_secrets/plugins/sendgrid.py

```diff
@@ -0,0 +1,17 @@
+"""
+This plugin searches for SendGrid API keys
+"""
+import re
+
+from detect_secrets.plugins.base import RegexBasedDetector
+
+
+class SendGridDetector(RegexBasedDetector):
+    """Scans for SendGrid API keys."""
+    secret_type = 'SendGrid API key'
+
+    denylist = [
+        # SendGrid API key
+        # ref. https://d2w67tjf43xwdp.cloudfront.net/Classroom/Basics/API/what_is_my_api_key.html
```

Comment:
In case `https://d2w67tjf43xwdp.cloudfront.net` looks odd to someone reading this, I tried Google-foo'ing and got the same. 

https://docs.sendgrid.com/ui/account-and-settings/api-keys doesn't give an example.